### PR TITLE
Refine `*strlen()` return type to exclude negative integers

### PR DESCRIPTION
**Pull request by staabm**
analog https://github.com/vimeo/psalm/commit/142ee57836c9b69299f3b2b701db56a4a7673eee

### CODE REVIEW

File: resources/functionMap.php [hunk 1]
```diff
@@ -9395,7 +9395,7 @@
 'Redis::sRem' => ['int', 'key'=>'string', 'member1'=>'string', '...other_members='=>'string'],
 'Redis::sRemove' => ['int', 'key'=>'string', 'member1'=>'string', '...other_members='=>'string'],
 'Redis::sScan' => ['array|bool', 'key'=>'string', '&iterator'=>'int', 'pattern='=>'string', 'count='=>'int'],
-'Redis::strLen' => ['int', 'key'=>'string'],
+'Redis::strLen' => ['0|positive-int', 'key'=>'string'],
```
Comment (staabm) [inline resources/functionMap.php:9398]: see https://github.com/phpredis/phpredis#strlen

**Comment by ondrejmirtes** [comment]
The tests here need fixing.

**Comment by staabm** [comment]
> The tests here need fixing.

sorry, locally on php8 it was green.

`mb_strlen` seems to have some special cases I don't want to dig into. reverted these parts for now.

on the initial commit these `mbStrlen*` cases failed on php 7.3 while were fine on php8.
https://github.com/phpstan/phpstan-src/runs/4951628678?check_suite_focus=true

fixing these for php7, made them fail on php8.
https://github.com/phpstan/phpstan-src/runs/4954732688?check_suite_focus=true

**Comment by ondrejmirtes** [comment]
mb_strlen has a dynamic return type extension: MbFunctionsReturnTypeExtension

**Comment by ondrejmirtes** [comment]
Thank you!

**Pull request merged**